x-pack/plugins/security_solution/public/common/components/empty_page/index.tsx

```diff
@@ -4,84 +4,122 @@
   align-self: center; /* Corrects horizontal centering in IE11 */
+  max-width: 60em;
 `;
 
 EmptyPrompt.displayName = 'EmptyPrompt';
 
+interface EmptyPageActions {
+  icon?: IconType;
+  label: string;
+  target?: string;
+  url: string;
+  descriptionTitle?: string;
+  description?: string;
+  fill?: boolean;
+  onClick?: MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+}
+
+export type EmptyPageActionsProps = Record<string, EmptyPageActions>;
+
 interface EmptyPageProps {
-  actionPrimaryIcon?: IconType;
-  actionPrimaryLabel: string;
-  actionPrimaryTarget?: string;
-  actionPrimaryUrl: string;
-  actionPrimaryFill?: boolean;
-  actionSecondaryIcon?: IconType;
-  actionSecondaryLabel?: string;
-  actionSecondaryTarget?: string;
-  actionSecondaryUrl?: string;
-  actionSecondaryOnClick?: MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+  actions: EmptyPageActionsProps;
   'data-test-subj'?: string;
   message?: ReactNode;
   title: string;
 }
 
-export const EmptyPage = React.memo<EmptyPageProps>(
-  ({
-    actionPrimaryIcon,
-    actionPrimaryLabel,
-    actionPrimaryTarget,
-    actionPrimaryUrl,
-    actionPrimaryFill = true,
-    actionSecondaryIcon,
-    actionSecondaryLabel,
-    actionSecondaryTarget,
-    actionSecondaryUrl,
-    actionSecondaryOnClick,
-    message,
-    title,
-    ...rest
-  }) => (
+const EmptyPageComponent = React.memo<EmptyPageProps>(({ actions, message, title, ...rest }) => {
+  const titles = Object.keys(actions);
+  const renderActions = useMemo(
+    () =>
+      Object.values(actions)
+        .filter((a) => a.label && a.url)
+        .map(
+          (
+            {
+              icon,
+              label,
+              target,
+              url,
+              descriptionTitle = false,
+              description = false,
+              onClick,
+              fill = true,
+            },
+            idx
+          ) =>
+            descriptionTitle != null || description != null ? (
+              <EuiFlexItem
+                grow={false}
+                style={{ maxWidth: 283 }}
+                key={`empty-page-${titles[idx]}-action`}
+              >
+                <EuiCard
+                  title={descriptionTitle}
+                  description={description}
+                  footer={
+                    /* eslint-disable-next-line @elastic/eui/href-or-on-click */
+                    <EuiButton
+                      href={url}
+                      onClick={onClick}
+                      iconType={icon}
+                      target={target}
+                      fill={fill}
+                      data-test-subj={`empty-page-${titles[idx]}-action`}
+                    >
+                      {label}
+                    </EuiButton>
+                  }
+                />
+              </EuiFlexItem>
+            ) : (
+              <EuiFlexItem
+                grow={false}
+                style={{ maxWidth: 283 }}
```

Comment:
Consider extracting `{ maxWidth: 283 }` into a `const` outside of this component to avoid repeating the magic number `238`, which first appears on line [68](https://github.com/elastic/kibana/pull/73050/files#diff-55072352674bf615172f2a3e02a49dbfR68) above.